salsa20/src/xsalsa.rs

```diff
@@ -136,3 +136,18 @@ pub fn hsalsa<R: Unsigned>(key: &Key, input: &Array<u8, U16>) -> Array<u8, U32>
 
     output
 }
+
+/// The Salsa20 quarter round function
+// for simplicity this function is copied from the software backend
+pub(crate) fn quarter_round(
+    a: usize,
+    b: usize,
+    c: usize,
+    d: usize,
+    state: &mut [u32; STATE_WORDS],
+) {
+    state[b] ^= state[a].wrapping_add(state[d]).rotate_left(7);
+    state[c] ^= state[b].wrapping_add(state[a]).rotate_left(9);
+    state[d] ^= state[c].wrapping_add(state[b]).rotate_left(13);
+    state[a] ^= state[d].wrapping_add(state[c]).rotate_left(18);
+}
```

Comment:
Why are you duplicating this?

Reply:
I tried to follow what is done at https://github.com/RustCrypto/stream-ciphers/blob/fea3dd013ee9c35fba56903ad44b411957de8cb2/chacha20/src/xchacha.rs#L147 iirc.

Reply:
Let me know if you will prefer to do it otherwise, i can change it up.

Reply:
The duplication there is less than ideal. If we can avoid it here that'd be great.

Reply:
Got it, thanks, i made the change at  https://github.com/RustCrypto/stream-ciphers/pull/348/commits/61cef0f6a36ae0c2821582e2523c9c70533d86c3

I can try to do the same in xchacha (in another PR) if you think it will worth it.

Reply:
Looks good now, thanks!

And that'd be great re: XChaCha